app/services/recipe_generator_service.rb

```diff
@@ -39,10 +40,20 @@ def request_messages
   def system_message
     [{ role: 'system', content: prompt }]
   end
-
+  
+  def preference_info
+    @preferences.map { |pref| "Consider this preference: #{pref.description}. Restriction: #{pref.restriction}" }
```

Comment:
With this you transform in a string like "Consider this preference: pref1. Restriction: rest2; pref2. Restrictoin: rest2, etc". Maybe is better this way for prompt instead of using an array.

```suggestion
    @preferences.map { |pref| "Consider this preference: #{pref.description}. Restriction: #{pref.restriction}" }.join('; ')
```